src/Agent/NewRelic/Agent/Extensions/NewRelic.Agent.Extensions/AwsSdk/SqsHelper.cs

```diff
@@ -61,17 +62,19 @@ public static ISegment GenerateSegment(ITransaction transaction, MethodCall meth
             return segment;
         }
 
-        public static void InsertDistributedTraceHeaders(ITransaction transaction, object sendMessageRequest)
+        public static void InsertDistributedTraceHeaders(ITransaction transaction, object sendMessageRequest, IEnumerable<string> dtHeaders)
         {
             var headersInserted = 0;
 
+            var requiredHeadroom = dtHeaders.Count();
```

Comment:
This call enumerates the IEnumerable it might be more efficient if we just had access to the List or IList instead.